d2/src/main/java/com/linkedin/d2/balancer/clusterfailout/ClusterFailoutConfig.java

```diff
@@ -0,0 +1,35 @@
+/*
+   Copyright (c) 2022 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package com.linkedin.d2.balancer.clusterfailout;
+
+import java.util.Set;
+
+public interface ClusterFailoutConfig
+{
+  /**
+   * Checks if this cluster is failed out.
+   * A failout can be scheduled to stop at a future time. When this happens, we will still have an active failout
+   * config and use the config to figure out if failout is finished.
+   *
+   * @return true if cluster is failed out.
+   */
+  boolean isFailedOut();
+
+  /**
+   * Gets all the configured peer clusters for accepting re-routed requests.
+   */
+  Set<String> getPeerClusters();
```

Comment:
Is this sufficient to handle the rerouting? I assume this interface implementation is telling the delegate how to subsequently perform a new discovery on another cluster?

Reply:
Specifically, at the pegasus level it seems we work with URIs for the request, so should we offload the URL building to be internal to this class?